libtextsecure/message_receiver.js

```diff
@@ -731,50 +854,7 @@ MessageReceiver.prototype.extend({
     }
 
     return promise
-      .then(async plaintext => {
-        const { isMe, isBlocked } = plaintext || {};
-        if (isMe || isBlocked) {
-          this.removeFromCache(envelope);
-          return null;
-        }
-
-        let conversation;
-        try {
-          conversation = await window.ConversationController.getOrCreateAndWait(
-            envelope.source,
-            'private'
-          );
-        } catch (e) {
-          window.log.info('Error getting conversation: ', envelope.source);
-        }
-
-        // Type here can actually be UNIDENTIFIED_SENDER even if
-        // the underlying message is FRIEND_REQUEST
-        if (
-          envelope.type !== textsecure.protobuf.Envelope.Type.FRIEND_REQUEST
-        ) {
-          // If we got here there is a valid session, which meants friend request
-          // is complete (if it wasn't already)
-          if (conversation) {
-            const isFriendRequestAccept = await conversation.onFriendRequestAccepted();
-            if (isFriendRequestAccept) {
-              await conversation.notifyFriendRequest(
-                envelope.source,
-                'accepted'
-              );
-            }
-          }
-        }
-
-        this.updateCache(envelope, plaintext).catch(error => {
-          window.log.error(
-            'decrypt failed to save decrypted message contents to cache:',
-            error && error.stack ? error.stack : error
-          );
-        });
-
-        return plaintext;
-      })
+      .then(plaintext => this.postDecrypt(envelope, plaintext))
```

Comment:
is the envelope (first arg) set when calling postDecrypt after the case  `textsecure.protobuf.Envelope.Type.PREKEY_BUNDLE` or `textsecure.protobuf.Envelope.Type.CIPHERTEXT`

Reply:
`envelope` is already set at the top of `async decrypt(envelope, ciphertext)` right?